hooks/open-telemetry/src/main/java/dev/openfeature/contrib/hooks/otel/OpenTelemetryHook.java

```diff
@@ -1,29 +1,69 @@
 package dev.openfeature.contrib.hooks.otel;
 
-import dev.openfeature.sdk.Client;
-import dev.openfeature.sdk.NoOpProvider;
-import dev.openfeature.sdk.OpenFeatureAPI;
+import dev.openfeature.sdk.FlagEvaluationDetails;
+import dev.openfeature.sdk.Hook;
+import dev.openfeature.sdk.HookContext;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.trace.Span;
 
-/** 
- * A placeholder.
+import java.util.Map;
+
+/**
+ * The OpenTelemetry hook  provides a way to automatically add a feature flag evaluation to a span as a span event.
+ * Refer to <a href="https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/feature-flags.md">OpenTelemetry</a>
  */
-public class OpenTelemetryHook {
-    
-    /** 
+public class OpenTelemetryHook implements Hook {
+
+    private final AttributeKey<String> flagKeyAttributeKey = AttributeKey.stringKey("feature_flag.flag_key");
+
+    private final AttributeKey<String> providerNameAttributeKey = AttributeKey.stringKey("feature_flag.provider_name");
+
+    private final AttributeKey<String> variantAttributeKey = AttributeKey.stringKey("feature_flag.variant");
+
+    private static final String EVENT_NAME = "feature_flag";
+
+    /**
      * Create a new OpenTelemetryHook instance.
      */
     public OpenTelemetryHook() {
     }
 
-    /** 
-     * A test method...
+    /**
+     * Records the event in the current span after the successful flag evaluation.
      *
-     * @return {boolean}
+     * @param ctx Information about the particular flag evaluation
+     * @param details Information about how the flag was resolved, including any resolved values.
+     * @param hints  An immutable mapping of data for users to communicate to the hooks.
      */
-    public static boolean test() {
-        OpenFeatureAPI.getInstance().setProvider(new NoOpProvider());
-        Client client = OpenFeatureAPI.getInstance().getClient();
-        return client.getBooleanValue("test2", true);
+    @Override
+    public void after(HookContext ctx, FlagEvaluationDetails details, Map hints) {
+        Span currentSpan = Span.current();
+        String value = details.getValue() != null ? details.getValue().toString() : "";
```

Comment:
As per the [semantic convention](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/feature-flags.md), we need to first use the `FlagEvaluationDetails.getVariant` to populate the `"feature_flag.variant"` and only fall back to the value if the variant is not set.

I'm also not sure about an empty string if the value is `null`. Is that better than `null`?

Reply:
Yes.. You're right. I interpret it in the wrong way. now, fixed it. 

> I'm also not sure about an empty string if the value is null. Is that better than null?

I just gave a second thought about this. If we convert the null value to empty string, it might convey different meaning to the user as well. so decided to keep is as null.

Reply:
Makes sense.